src/test/java/io/gravitee/policy/cache/CachePolicyV4EmulationEngineIntegrationTest.java

```diff
@@ -293,6 +298,57 @@ void shouldBypassCache_UsingParams(HttpClient client) throws Exception {
         DummyCacheResource.checkNumberOfCacheEntries(0);
     }
 
+    @Test
+    @DisplayName("Should invoke cache instead of backend when doing the same call twice and should encode/decode content")
+    void shouldUseCache_BinaryContentType(HttpClient client) throws Exception {
+        wiremock.stubFor(get("/endpoint").willReturn(ok(RESPONSE_FROM_BACKEND_1).withHeader("Content-Type", "image/png")));
+
+        final var secondObs = client
+            .rxRequest(HttpMethod.GET, "/test")
+            .flatMap(HttpClientRequest::rxSend)
+            .flatMapPublisher(response -> {
+                assertThat(response.statusCode()).isEqualTo(200);
+                return response.toFlowable();
+            })
+            .test();
+
+        secondObs.await(1000, TimeUnit.MILLISECONDS);
+        secondObs
+            .assertComplete()
+            .assertValue(buffer -> {
+                assertThat(buffer).hasToString(RESPONSE_FROM_BACKEND_1);
+                return true;
+            })
+            .assertNoErrors();
+
+        wiremock.stubFor(get("/endpoint").willReturn(ok(RESPONSE_FROM_BACKEND_2).withHeader("Content-Type", "application/json")));
+
+        client
+            .rxRequest(HttpMethod.GET, "/test")
+            .flatMap(HttpClientRequest::rxSend)
+            .flatMapPublisher(response -> {
+                assertThat(response.statusCode()).isEqualTo(200);
+                return response.toFlowable();
+            })
+            .test();
```

Comment:
You mentioned 2 consecutive calls, is this call mandatory?

Reply:
yes the second call retrieves the data from the cache, so just to make sure the data is decoded